CFC_DataCollector/recommender/trip.py

```diff
@@ -0,0 +1,109 @@
+#maps team provided get_cost function
+#from common.featurecalc import get_cost
+import jsonpickle
+
+
+# def unit_test():
+#     info = open("C:/Users/rzarrabi/Documents/school/junior/spring/trip/e-mission-server/CFC_DataCollector/tests/data/missing_trip", "r")
+#     t = Trip(info.read())
+#     return tt
+
+class Trip(object): 
+    #Instance parameters
+    def __init__(self, json_segment):
+        self.trip_from_json(json_segment)
+
+    def trip_from_json(self, json_seg):
```

Comment:
I beleive the way it works is that you call Trip(json) and that creates a
new trip

On Sun, Mar 29, 2015 at 9:51 PM, shankari <notifications@github.com> wrote:

> In CFC_DataCollector/recommender/trip.py
> <https://github.com/e-mission/e-mission-server/pull/103#discussion_r27366627>
> :
>
> > +#maps team provided get_cost function
> > +#from common.featurecalc import get_cost
> > +import jsonpickle
> > +
> > +
> > +# def unit_test():
> > +#     info = open("C:/Users/rzarrabi/Documents/school/junior/spring/trip/e-mission-server/CFC_DataCollector/tests/data/missing_trip", "r")
> > +#     t = Trip(info.read())
> > +#     return tt
> > +
> > +class Trip(object):
> > +    #Instance parameters
> > +    def __init__(self, json_segment):
> > +        self.trip_from_json(json_segment)
> > +
> > +    def trip_from_json(self, json_seg):
>
> @joshzarrabi <https://github.com/joshzarrabi>, @sdsingh
> <https://github.com/sdsingh>,
> Shouldn't this be a staticmethod that returns the appropriate trip
> subclass? Otherwise, how will the TripIterator work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/e-mission/e-mission-server/pull/103/files#r27366627>.
>